service/src/test/java/uk/gov/hmcts/reform/fpl/service/RespondentServiceTest.java

```diff
@@ -0,0 +1,259 @@
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.RespondentParty;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(SpringExtension.class)
+class RespondentServiceTest {
+
+    private final RespondentService service = new RespondentService();
+
+    @Test
+    void shouldAddMigratedRespondentYesWhenNoRespondentData() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(createData("data", "some data"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = service.setMigratedValue(caseDetails);
+
+        assertThat(response.getData()).containsEntry("respondentsMigrated", "Yes");
+    }
+
+    @Test
+    void shouldAddMigratedRespondentYesWhenRespondents1Exists() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(createData("respondents1", "some value"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = service.setMigratedValue(caseDetails);
+
+        assertThat(response.getData()).containsEntry("respondentsMigrated", "Yes");
+    }
+
+    @Test
+    void shouldAddMigratedRespondentNoWhenOldRespondentsExists() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(createData("respondents", "some value"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = service.setMigratedValue(caseDetails);
+
+        assertThat(response.getData()).containsEntry("respondentsMigrated", "No");
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void shouldAddPartyIDAndPartyTypeValuesToSingleRespondent() {
+        Map<String, Object> respondentObject = new HashMap<>();
+
+        respondentObject.put("respondents1", ImmutableList.of(
+            ImmutableMap.of(
+                "id", "12345",
+                "value", ImmutableMap.of(
+                    "party", RespondentParty.builder()
+                        .firstName("James")
+                        .build()
+                ))));
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(respondentObject)
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = service.addHiddenValues(caseDetails);
+
+        Map<String, Object> data = response.getData();
+        List<Map<String, Object>> respondents = (List<Map<String, Object>>) data.get("respondents1");
+        Map<String, Object> value = (Map<String, Object>) respondents.get(0).get("value");
+        Map<String, Object> party = (Map<String, Object>) value.get("party");
+
+        assertThat(party)
+            .containsEntry("firstName", "James")
+            .containsEntry("partyType", "INDIVIDUAL");
+
+        assertThat(party.get("partyID")).isNotNull();
+    }
+
+    @Disabled
+    @SuppressWarnings("unchecked")
+    @Test
+    void shouldMapParameterNamesToConstructorArguments() {
```

Comment:
I do not get what this test is for and it is disabled. Please remove if not needed anymore.

Reply:
this was the test that was failing - I am concious that my tests do not accurately reflect how the data will come into the system, as I am using builders to construct my data. Thoughts. Happy to remove. It is failing due to mapping exceptions